### PR TITLE
Revamps the randomized Hub 01 mercenary into a defined character

### PR DESCRIPTION
**Pull request by Ilysen**
#### Summary
Content "The randomized Hub 01 mercenary is now a defined character, with a new name, personality, and background"

#### Purpose of change

Pretty much everything in my initial PR to create Smokes can be repeated here; in addition to wanting to update the dialogue and polish out kinks, I think that consistently-appearing characters/static NPCs are much more interesting when they're genuine characters with defined personalities and behavior.

#### Describe the solution

The new NPC is named Cranberry Foster, because I couldn't think of anything else for the first name. They're a former park ranger with blond hair, blue eyes, and rose skin. Personality-wise, they're very laid back and a little awkward, but generally patient and always genuine.

Code changes:
* Renamed `NPC_ROBOFAC_MERC_1.json` to `NPC_Cranberry_Foster.json`. All Git history is preserved in the new file.
* Stats, skills, and appearance are now fixed, instead of being random. Being a park ranger, they have a wide variety of skills, as well as being quite skilled in food handling, computers, and electronics. Pretty competent survivor overall.
* Now wears a wedding ring.
* Now spawns with 7 protein rations, 2 bottles of clean water, 15-18 Hub 01 gold coins, a smart phone, and a first aid kit. This is all for flavor, for the most part.
* Replaced the repeatable mission to bring 200 units of gold for 8 Hub 01 coins with a dialogue option that does the same thing, for the purpose of cleanliness and allowing players to trade easily in batch. The mission def has been moved to `obsolete.json`.

Dialogue changes:
* Introduced unique dialogue when meeting the NPC for the first time.
* Rewrote almost all existing dialogue responses to match the new personality and characterization.
* Now features contextual responses when asking if they know anything about Hub 01.
* Questions asking about the HWP and how to get it are removed from the dialogue options when Cranberry promises to vouch for the player in obtaining one.
* Cranberry's gender and sex are never defined or hinted at, as their face and voice are both androgynous. Implemented a line of Cranberry teasing the player for asking, which becomes available once trust has reached a certain threshold.
* Implemented a long, one-time dialogue chain detailing parts of Cranberry's past life and character that become available once the player has helped them with their mission or traded gold with them at least once.

#### Describe alternatives you've considered

Considered having the NPC spawn in as part of an EOC that triggers when the player is returning to Hub 01 after being away instead of immediately after finishing `Steal a dead man's mind`, to try and flesh out the sensation of things happening while they weren't there. I'm still open to this, and I might do it in this PR, but I wanted to get the meat and potatoes of the change up onto Github before I made smaller tweaks.

Writing is relative, so I'm open to suggestions.

#### Testing

Spawned in Hub 01 using the overmap editor and progressed the questline using spawned-in items until the NPC spawned. Walked through all of the dialogue options; verified that gold trading was working, as well as dialogue options changing or appearing as Cranberry's trust with the player grew.

#### Additional context

image

image

image

image

### CODE REVIEW

**Comment by Ilysen** [comment]
CC: @John-Candlebury, since Hub 01 is kind of their thing

**Comment by Gotdamnmiracle** [comment]
This looks like something I'd like to do for the arsonist character in the refugee center. I was hoping to flesh them out a bit more, but have zero idea how. 

I shot you a message on the CDDA discourse forum. Would you be able to walk me through the process?

Additionally, if you'd like to write up more NPCs I'd love to toss my hat in the ring.

**Comment by Ilysen** [comment]
I haven't actually used Discourse in ages and don't have the credentials, so I wouldn't expect a response there. I'm a pretty terrible teacher, but I can refer you to the following PRs as reference material:

#56438
#56308 
#52374

File: data/json/npcs/robofac/NPC_Cranberry_Foster.json [hunk 1]
```diff
@@ -0,0 +1,803 @@
...
+  },
+  {
+    "id": "TALK_ROBOFAC_MERC_1",
+    "type": "talk_topic",
+    "dynamic_line": {
+      "u_has_var": "u_met_Cranberry",
+      "type": "general",
+      "context": "meeting",
+      "value": "yes",
+      "no": "&There's a person reclined on the sofa here.  They're very tall, with their legs reaching to the other armrest, and they're wearing a set of Hub 01's modular armor.  An environmental suit in blue-and-brown is crumpled in a misshapen heap on the floor near the table, and leaning against the far armrest is a rifle looking like nothing you've seen before.\n\nThey look up towards you as you come through the door and wave a hand.  \"Placeholder text.\"",
```
Comment (Inglonias) [inline data/json/npcs/robofac/NPC_Cranberry_Foster.json:107]: ah yes, the most common greeting after the cataclysm: "placeholder text".

Comment (Ilysen) [reply data/json/npcs/robofac/NPC_Cranberry_Foster.json:107]: It's deliberate - see the dialogue that directly has the player responding to it. :p

**Comment by Ilysen** [comment]
I haven't heard any concerns or discussion about this so far, so it should be ready for review and possibly merge (if everything seems ok).

**Comment by Inglonias** [comment]
The minor spelling error I discovered during my review has not been resolved yet.

**Comment by Ilysen** [comment]
What would that be? Your comment only makes a remark about what the character says in their intro text, not any kind of grammar concerns.

**Comment by Inglonias** [comment]
> What would that be? Your comment only makes a remark about what the character says in their intro text, not any kind of grammar concerns.

There was a second comment regarding line 389 of NPC_Cranberry_Foster.json. They say `Uh, call be Cranberry`, when it should be `Uh, call me Cranberry`. Sorry. Github is being weird.

**Comment by Ilysen** [comment]
Good eye, thanks.

**Pull request merged**